src/test/java/com/redhat/jenkins/plugins/ci/integration/AmqMessagingPluginIntegrationTest.java

```diff
@@ -421,6 +421,33 @@ public void testSimpleCIEventTriggerWithParamOverride() throws Exception, Interr
     }
 
     @Test
+    public void testSimpleCIEventTriggerHeadersInEnv() throws Exception, InterruptedException {
+        FreeStyleJob jobA = jenkins.jobs.create();
+        jobA.configure();
+        CIEventTrigger ciEvent = new CIEventTrigger(jobA);
+        ciEvent.selector.set("CI_TYPE = 'code-quality-checks-done'");
+
+        // We are only checking that this shows up in the console output.
+        jobA.addShellStep("echo $CI_HEADERS");
+        jobA.save();
+
+        FreeStyleJob jobB = jenkins.jobs.create();
+        jobB.configure();
+        CINotifierPostBuildStep notifier = jobB.addPublisher(CINotifierPostBuildStep.class);
+
+        notifier.messageType.select("CodeQualityChecksDone");
+        notifier.messageContent.set("some irrelevant content");
+
+        jobB.save();
+        jobB.startBuild().shouldSucceed();
+
+        elasticSleep(1000);
+        jobA.getLastBuild().shouldSucceed().shouldExist();
+        assertThat(jobA.getLastBuild().getConsole(), containsString("\"CI_STATUS\":\"passed\""));
+        assertThat(jobA.getLastBuild().getConsole(), containsString("\"CI_TYPE\":\"code-quality-checks-done\""));
```

Comment:
OK, I'm not sure how to improve this any further.

The value of `CI_HEADERS` looks like:

```json
{"CI_STATUS":"passed","CI_NAME":"efficient_basis","CI_TYPE":"code-quality-checks-done"}
```
I'm currently checking for `CI_STATUS` and `CI_TYPE`.  I'm not sure how to check for the full string though since `CI_NAME` is randomly generated (or if its worth it to work on it any further).  What do you think @scoheb?